Amplify/Categories/DataStore/Model/Temporal/TimeZone+Extension.swift

```diff
@@ -0,0 +1,136 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+
+import Foundation
+
+extension TimeZone {
+    private static let iso8601TimeZoneHHColonMMColonSSRegex = try? NSRegularExpression(pattern: "^[+-]\\d{2}:\\d{2}:\\d{2}$")
+    private static let iso8601TimeZoneHHColonMMRegex = try? NSRegularExpression(pattern: "^[+-]\\d{2}:\\d{2}$")
+    private static let iso8601TimeZoneHHMMRegex = try? NSRegularExpression(pattern: "^[+-]\\d{2}\\d{2}$")
+    private static let iso8601TimeZoneHHRegex = try? NSRegularExpression(pattern: "^[+-]\\d{2}$")
```

Comment:
As discussed around the maintainability of these regex, let's have some documentation around these

```suggestion

    /// The following regex's are used for parsing and validating time zone representations
    /// in ISO 8601 format. These patterns cover various combinations of hours, minutes,
    /// and seconds in the timezone offset.
    ///
    /// - `iso8601TimeZoneHHColonMMColonSSRegex`: Matches a time zone offset in the format
    ///   HH:MM:SS with an optional leading positive or negative sign. While ISO 8601
    ///   defines the offset with only hours and minutes, this regex includes seconds
    ///   for flexibility in our implementation. This is to support decoding payloads from with
    ///   services such as [AWS AppSync that provide the AWSDateTime type.](https://docs.aws.amazon.com/appsync/latest/devguide/scalars.html).
    ///
    /// - `iso8601TimeZoneHHColonMMRegex`:  HH:MM with sign.
    ///
    /// - `iso8601TimeZoneHHMMRegex`: HHMM with sign.
    ///
    /// - `iso8601TimeZoneHHRegex`:  HH with sign.
    private static let iso8601TimeZoneHHColonMMColonSSRegex = try? NSRegularExpression(pattern: "^[+-]\\d{2}:\\d{2}:\\d{2}$")
    private static let iso8601TimeZoneHHColonMMRegex = try? NSRegularExpression(pattern: "^[+-]\\d{2}:\\d{2}$")
    private static let iso8601TimeZoneHHMMRegex = try? NSRegularExpression(pattern: "^[+-]\\d{2}\\d{2}$")
    private static let iso8601TimeZoneHHRegex = try? NSRegularExpression(pattern: "^[+-]\\d{2}$")
```